client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { Button, Gridicon } from '@automattic/components';
+import { Fragment, useCallback } from '@wordpress/element';
+import classNames from 'classnames';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import CardHeading from 'calypso/components/card-heading';
+import { MailboxForm } from 'calypso/my-sites/email/form/mailboxes';
+import { MailboxFormWrapper } from 'calypso/my-sites/email/form/mailboxes/components/mailbox-form-wrapper';
+import { MailboxOperations } from 'calypso/my-sites/email/form/mailboxes/components/utilities/mailbox-operations';
+import { sanitizeMailboxValue } from 'calypso/my-sites/email/form/mailboxes/components/utilities/sanitize-mailbox-value';
+import {
+	FIELD_FIRSTNAME,
+	FIELD_MAILBOX,
+	FIELD_NAME,
+} from 'calypso/my-sites/email/form/mailboxes/constants';
+import {
+	EmailProvider,
+	MailboxFormFieldBase,
+	MutableFormFieldNames,
+} from 'calypso/my-sites/email/form/mailboxes/types';
+
+import './style.scss';
+
+interface MailboxListProps {
+	areButtonsBusy?: boolean;
+	hiddenFieldNames?: MutableFormFieldNames[];
+	onCancel?: () => void;
+	onSubmit: ( mailboxOperations: MailboxOperations ) => void;
+	provider: EmailProvider;
+	selectedDomainName: string;
+	showAddNewMailboxButton?: boolean;
+	showCancelButton?: boolean;
+	submitActionText?: TranslateResult;
+}
+
+const NewMailBoxList = ( props: MailboxListProps & { children?: JSX.Element } ): JSX.Element => {
+	const translate = useTranslate();
+
+	const {
+		areButtonsBusy = false,
+		hiddenFieldNames = [],
+		onCancel = () => undefined,
+		onSubmit,
+		provider,
+		selectedDomainName,
+		showAddNewMailboxButton = false,
+		showCancelButton = false,
+		submitActionText = translate( 'Submit' ),
+	} = props;
+
+	const createNewMailbox = () => new MailboxForm< EmailProvider >( provider, selectedDomainName );
+
+	const [ mailboxes, setMailboxes ] = useState( [ createNewMailbox() ] );
+
+	// Set visibility for desired hidden fields
+	mailboxes.forEach( ( mailbox ) => {
+		hiddenFieldNames.forEach( ( fieldName ) => {
+			mailbox.setFieldIsVisible( fieldName, false );
+			mailbox.setFieldIsRequired( fieldName, false );
+		} );
+	} );
+
+	const addMailbox = () => {
+		setMailboxes( [ ...mailboxes, createNewMailbox() ] );
+	};
+
+	const removeMailbox = useCallback(
+		( uuid: string ) => () => {
+			setMailboxes( mailboxes.filter( ( mailbox ) => mailbox.formFields.uuid.value !== uuid ) );
+		},
+		[ mailboxes ]
+	);
+
+	const handleCancel = () => onCancel();
+
+	const persistMailboxesToState = () => {
+		setMailboxes( [ ...mailboxes ] );
+	};
+
+	const handleSubmit = () => {
+		mailboxes.forEach( ( mailbox ) => mailbox.validate( true ) );
+		persistMailboxesToState();
+		onSubmit( new MailboxOperations( mailboxes, persistMailboxesToState ) );
+	};
+
+	return (
+		<div className="new-mailbox-list__main">
```

Comment:
I'm thinking that this could be an actual `<form>` element with an `onSubmit` handler. Should be helpful both from a UX perspective and accessibility perspective.

Reply:
Yeah, I actually thought about this for some time. Although it seems as if we have been mostly evading `<form>`  for some reason.

However I would love to handle this in another PR as this is big enough.

Reply:
I've noticed that too 🤔 Sure, let's save it for later then!

Reply:
Decided to add it afterall.